src/test/java/org/mockbukkit/mockbukkit/inventory/meta/ItemMetaMockTest.java

```diff
@@ -1604,6 +1643,39 @@ void givenPossibleValues(ItemRarity value)
 			assertEquals(value, meta.getRarity());
 		}
 
+		@Test
+		void influencesHashCode()
+		{
+			Set<Integer> hashcodes = new HashSet<>();
+			meta.setRarity(null);
+			hashcodes.add(meta.hashCode());
+			for (ItemRarity rarity : ItemRarity.values())
+			{
+				meta.setRarity(rarity);
+				assertTrue(hashcodes.add(meta.hashCode()));
+			}
+		}
+
+		@Test
+		void valueIsCloned()
+		{
+			for (ItemRarity rarity : ItemRarity.values())
+			{
+				meta.setRarity(rarity);
+				assertEquals(rarity, meta.clone().getRarity());
+			}
+		}
```

Comment:
Please replace this with Parameterized test, so that its easier to identify the faulty rarity.